packages/matter.js/src/device/Endpoint.ts

```diff
@@ -141,7 +141,7 @@ export class Endpoint {
         if (cluster.id === DescriptorCluster.id) {
             this.descriptorCluster = cluster as unknown as ClusterServerObjForCluster<typeof DescriptorCluster>;
         }
-        this.clusterServers.set(cluster.id, cluster);
+        this.clusterServers.set(cluster.id, cluster as any);
```

Comment:
why? do we know ahat changed in TS to need that now? Whats the exact error?

Reply:
Gives the "types do not have sufficient overlap" error.  Seems like it should be OK without the cast...  May be a TS5 bug but it happened on both 5.0 and 5.2.

I'll add comments to this effect, that sufficient?

Reply:
I wonder why he accepts any because especially this shozld not overkllap with

> private readonly clusterServers = new Map<ClusterId, ClusterServerObj<Attributes, Events>>();

Reply:
because in the case you say I would have expected an "as unknown as Bla"

Reply:
`any` cast implicitly to any type.  I usually only use `as unknown as Type` when the expected type isn't already known, like below where there's a return without a return type annotation.

Reply:
We can check later ... might really be a TS5 thing :-(

Reply:
I updated the comment with more detail.  Resolving for now per Discord conversation